testrunner-running/src/main/scala/edu/illinois/cs/statecapture/CustomAbstractReflectionConverter.java

```diff
@@ -0,0 +1,53 @@
+package edu.illinois.cs.statecapture;
+
+import com.thoughtworks.xstream.converters.ConversionException;
+import com.thoughtworks.xstream.converters.UnmarshallingContext;
+import com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter;
+import com.thoughtworks.xstream.converters.reflection.ReflectionProvider;
+import com.thoughtworks.xstream.io.HierarchicalStreamReader;
+import com.thoughtworks.xstream.mapper.Mapper;
+
+import java.lang.reflect.Field;
+
+public abstract class CustomAbstractReflectionConverter extends AbstractReflectionConverter {
+
+    public CustomAbstractReflectionConverter(final Mapper mapper, final ReflectionProvider reflectionProvider) {
+        super(mapper, reflectionProvider);
+    }
+
+    @Override
+    public Object doUnmarshal(final Object result, final HierarchicalStreamReader reader,
+                              final UnmarshallingContext context) {
+        if (!UnmarshalChain.isInitialized()) {
```

Comment:
Do we need to initialize specifically using system properties, considering that we have logic that initializes the chain through its constructor? Is there ever a chance that the chain is not initialized properly (and the system properties are set correctly)?